lib/node_modules/@stdlib/blas/ext/base/ssumpw/src/main.c

```diff
@@ -0,0 +1,135 @@
+*
+* ## References
+*
+* -   Higham, Nicholas J. 1993. "The Accuracy of Floating Point Summation." _SIAM Journal on Scientific Computing_ 14 (4): 783–99. doi:[10.1137/0914050](https://doi.org/10.1137/0914050).
+*
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  stride length
+* @return         output value
+*/
+double API_SUFFIX(stdlib_strided_ssumpw)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX ) {
+	CBLAS_INT ox = stdlib_strided_stride2offset( N, strideX );
+	API_SUFFIX(stdlib_strided_ssumpw_ndarray)( N, X, strideX, ox );
+}
+
+/**
+* Computes the sum of single-precision floating-point strided array elements using pairwise summation and alternative indexing semantics.
+*
+* ## Method
+*
+* -   This implementation uses pairwise summation, which accrues rounding error `O(log2 N)` instead of `O(N)`. The recursion depth is also `O(log2 N)`.
+*
+* ## References
+*
+* -   Higham, Nicholas J. 1993. "The Accuracy of Floating Point Summation." _SIAM Journal on Scientific Computing_ 14 (4): 783–99. doi:[10.1137/0914050](https://doi.org/10.1137/0914050).
+*
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  stride length
+* @param offsetX  starting index
+* @return         output value
+*/
+double API_SUFFIX(stdlib_strided_ssumpw_ndarray)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX ) {
+	CBLAS_INT ix;
+	CBLAS_INT M;
+	CBLAS_INT n;
+	CBLAS_INT i;
+	double sum;
+	double s0;
+	double s1;
+	double s2;
+	double s3;
+	double s4;
+	double s5;
+	double s6;
+	double s7;
+
+	if ( N <= 0 ) {
+		return 0.0f;
+	}
+	ix = offsetX;
+	if ( strideX == 0 ) {
+		if ( stdlib_base_is_nanf( X[ ix ] ) ) {
+			return 0.0f;
+		}
+		return N * X[ ix ];
+	}
+	if ( N < 8 ) {
+		// Use simple summation...
+		sum = 0.0f;
+		for ( i = 0; i < N; i++ ) {
+			sum += X[ ix ];
+			ix += strideX;
+		}
+		return sum;
+	}
+	// Blocksize for pairwise summation: 128 (NOTE: decreasing the blocksize decreases rounding error as more pairs are summed, but also decreases performance. Because the inner loop is unrolled eight times, the blocksize is effectively `16`.)
+	if ( N <= 128 ) {
+		// Sum a block with 8 accumulators (by loop unrolling, we lower the effective blocksize to 16)...
+		s0 = X[ ix ];
+		s1 = X[ ix+strideX ];
+		s2 = X[ ix+(2*strideX) ];
+		s3 = X[ ix+(3*strideX) ];
+		s4 = X[ ix+(4*strideX) ];
+		s5 = X[ ix+(5*strideX) ];
+		s6 = X[ ix+(6*strideX) ];
+		s7 = X[ ix+(7*strideX) ];
+		ix += 8 * strideX;
+
+		M = N % 8;
+		for ( i = 8; i < N-M; i += 8 ) {
+			s0 += X[ ix ];
+			s1 += X[ ix+strideX ];
+			s2 += X[ ix+(2*strideX) ];
+			s3 += X[ ix+(3*strideX) ];
+			s4 += X[ ix+(4*strideX) ];
+			s5 += X[ ix+(5*strideX) ];
+			s6 += X[ ix+(6*strideX) ];
+			s7 += X[ ix+(7*strideX) ];
+			ix += 8 * strideX;
+		}
+		// Pairwise sum the accumulators:
+		sum = ((s0+s1) + (s2+s3)) + ((s4+s5) + (s6+s7));
```

Comment:
```suggestion
		sum = ( (s0+s1) + (s2+s3) ) + ( (s4+s5) + (s6+s7) );
```

Reply:
It feels more readable without the spaces, WDYT? I suppose that's why they weren't there in the [initial implementation](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/ext/base/ssumpw/src/ssumpw.c#L101) as well.